### ISSUE TITLE
Make it compatible with v3 of Zod

### ISSUE THREAD

**Issue by mrhammadasif** [StefanTerdell/zod-to-json-schema#1]
As Zod is now shifted to v3 - Stable, we should make it compatible with the new version

Currently, following type error is being returned

```js
Argument of type 'ZodType<T, ZodTypeDef, T>' is not assignable to parameter of type 'ZodType<any, ZodTypeDef>'.
  Property 'toJSON' is missing in type 'ZodType<T, ZodTypeDef, T>' but required in type 'ZodType<any, ZodTypeDef>'
```

**Comment by StefanTerdell** [comment]
Yup, definately time to get on this. I will be implementing some changes to the Intersection parsing and releasing the v1 asap, after that I'll focus on V3. I do have quite a bit of regular work to do at the moment so not sure about the timeline

**Comment by mrhammadasif** [comment]
I have migrated, tested and created a PR #2 to support latest version (as of today which is zod@3.1.0)

**Comment by mrhammadasif** [comment]
@StefanTerdell any update on this?

**Comment by StefanTerdell** [comment]
Just released your updates as 3.0.1! Added you as contributor. Also once again, sorry for the delay. Let me know if there are any issues. Closing.
